lib/wallaroo/core/initialization/local_topology.pony

```diff
@@ -1474,6 +1498,22 @@ actor LocalTopologyInitializer is LayoutInitializer
     _outgoing_boundaries = consume bs
     _outgoing_boundary_builders = consume bbs
 
+  fun ref local_topology_for_joining_worker(
```

Comment:
I think you should pass the `LocalTopology` in here so that this is a pure and total function.  Do the match at the caller and Fail() there if it doesn't exist.